real_time/surface_src/java_src/Alice/src/test/java/com/roboclub/robobuggy/main/RobobuggyConfigFileTest.java

```diff
@@ -0,0 +1,216 @@
+        assert RobobuggyConfigFile.getRobotType().equals("TempBuggy!!!");
+    }
+
+    /**
+     * Creates a config file that specifies a TransistorAuton robot, then creates the robot
+     */
+    @Test
+    public void testTransistorAuton(){
+        try {
+            createConfigFileWithRobot("TransistorAuton");
+            RobobuggyConfigFile.loadConfigFile();
+        } catch (IOException e) {
+            e.printStackTrace();
+            assert false;
+        } catch (IllegalArgumentException e) {
+            e.printStackTrace();
+            assert false;
+        }
+        assert RobobuggyConfigFile.getRobot() instanceof TransistorAuton;
+    }
+
+    /**
+     * Creates a config file that specifies a TransistorDataCollection robot, then creates the robot
+     */
+    @Test
+    public void testTransistorDataCollection(){
+        try {
+            createConfigFileWithRobot("TransistorDataCollection");
+            RobobuggyConfigFile.loadConfigFile();
+        } catch (IOException e) {
+            e.printStackTrace();
+            Assert.fail();
+        } catch (IllegalArgumentException e) {
+            e.printStackTrace();
+            Assert.fail();
+        }
+        assert RobobuggyConfigFile.getRobot() instanceof TransistorDataCollection;
+    }
+
+    /**
+     * Creates a config file that specifies a PlayBackRobot robot, then creates the robot
+     */
+    @Test
+    public void testPlayBackRobot(){
+        try {
+            createConfigFileWithRobot("PlayBackRobot");
+            RobobuggyConfigFile.loadConfigFile();
+        } catch (IOException e) {
+            e.printStackTrace();
+            Assert.fail();
+        } catch (IllegalArgumentException e) {
+            e.printStackTrace();
+            Assert.fail();
+        }
+        assert RobobuggyConfigFile.getRobot() instanceof PlayBackRobot;
+    }
+
+    /**
+     * Creates a config file that specifies a SimRobot robot, then creates the robot
+     */
+    @Test
+    public void testSimRobot(){
+        try {
+            createConfigFileWithRobot("SimRobot");
+            RobobuggyConfigFile.loadConfigFile();
+        } catch (IOException e) {
+            e.printStackTrace();
+            Assert.fail();
+        } catch (IllegalArgumentException e) {
+            e.printStackTrace();
+            Assert.fail();
+        }
+        assert RobobuggyConfigFile.getRobot() instanceof SimRobot;
+    }
+
+    /**
+     * Creates a config file with a robot field but no robot name
+     * It's expected to throw an IllegalArgumentException
+     */
+    @Test
+    public void testNoName(){
+        try {
+            createConfigFileWithRobot("");
+            RobobuggyConfigFile.loadConfigFile();
+            Assert.fail();
+        } catch (IOException e) {
+            e.printStackTrace();
+            Assert.fail();
+        } catch (IllegalArgumentException e) {
+            //Good;test passed
+        }
```

Comment:
You need to add an `Assert.fail()` after your catch. Doing nothing here is the wrong thing. When you do that, inside the Illegal argument `catch` block, put a `return` statement there so you don't accidentally trigger failure